### PR TITLE
Run rustfmt on libarena.

### PR DESCRIPTION
**Pull request by ahmedcharles**

### CODE REVIEW

File: src/libarena/lib.rs [hunk 1]
```diff
@@ -297,7 +290,8 @@ impl<'longer_than_self> Arena<'longer_than_self> {
 
         unsafe {
             let buf = head.as_ptr();
-            (buf.offset(tydesc_start as isize), buf.offset(start as isize))
```
Comment (cristicbz) [inline src/libarena/lib.rs]: This is strange, it looks like it fits in 100 columns.

File: src/libarena/lib.rs [hunk 2]
```diff
@@ -545,14 +537,21 @@ mod tests {
 
     #[test]
     fn test_arena_alloc_nested() {
-        struct Inner { value: u8 }
-        struct Outer<'a> { inner: &'a Inner }
-        enum EI<'e> { I(Inner), O(Outer<'e>) }
+        struct Inner {
```
Comment (cristicbz) [inline src/libarena/lib.rs]: Do we really want to disallow this in the single field case?

Comment (nrc) [inline src/libarena/lib.rs]: There is a rustfmt issue for this already, but in the short term, yes it's fine

File: src/libarena/lib.rs (hunk 1 above)
Comment (nrc) [inline src/libarena/lib.rs]: There's a heuristic where we cut off into vertical formatting before we hit the absolute limit. We probably shouldn't do it for tuples though (only for function calls).

https://github.com/nrc/rustfmt/issues/450

**Comment by nrc** [comment]
@bors: r+ rollup

Thanks for the PR!

**Pull request merged**